lib/mail/fields/content_type_field.rb

```diff
@@ -144,10 +144,9 @@ def sanatize( val )
 
       # TODO: check if there are cases where whitespace is not a separator
       val = val.
+        gsub(/\s[\w\-]+:\s.*\\r\\n$/, ''). # cases with botched fields like Content-Type: text/html Content-Transfer-Encoding: 8bit\r\n
```

Comment:
Pardon, we haven't met. I don't know how you're using this library or what you need from it, so you're leaving me at a loss with the weight of your concerns and expectations.

Here's some background on the vision of this library, since you're implying that it's a strict parser, which is definitively is not.

The Mail library follows [Postel's classic principle](https://en.wikipedia.org/wiki/Robustness_principle): "Be conservative in what you do, be liberal in what you accept from others." We have a long history of liberal parsing to fulfill our aim to be very user-friendly to developers.

The Mail library should generally parse just about any email in the wild. You can see this demonstrated in our rake tasks that successfully parse a huge corpus of spam email.

This gets to the balance I discuss above. How liberal can we be without compromising valid email parsing (by introducing new bugs!) and taking on undue code complexity by littering our codebase with workarounds for rare, exceptional emails? Broken charsets and encodings are a great example. We go to great lengths to transcode to Ruby string encodings, but what is the right choice when we cannot? Should we make a best guess, force the encoding, raise an exception? These are policy decisions that one user may find as a blessing where another user finds them as a curse.

Our best option is to be reasonable and sensible about striking a balance. We can attempt to accept all the emails we can without making outsized compromises for very rare or exceptional cases. If you fall on the other side of this decisionmaking because you have such an exceptional email and feel the built-in parser should correct it—even with a single-case workaround that threatens new bugs and parsing errors with other valid emails—then it'll always be hard to live with this tradeoff. But: I hope we can agree it's a necessary and healthy tradeoff to make.

Please consider sharing your regression suite (with personal/private info scrubbed). Perhaps we can address other brokenness and help relieve your workaround burden from maintaining a fork.